### PR TITLE
Fix/middleware

### PR DESCRIPTION
**Pull request by AashishDhakal**

### CODE REVIEW

File: webhooks/middleware/protocols/soap.py [hunk 1]
```diff
@@ -16,6 +16,9 @@ def forward_to_SOAP_service(json_data, image):
     timestamp = json_data["data"]["timestamp_local"]
     plate = json_data["data"]["results"][0]["plate"]
     score = json_data["data"]["results"][0]["score"]
```
Comment (ttayson) [inline webhooks/middleware/protocols/soap.py:18]:
```suggestion
    score = json_data["data"]["results"][0].get("score")
```

In vehicle mode the score is not sent on this key, I'm doing another integration and I noticed this error.

File: webhooks/middleware/protocols/soap.py [hunk 2]
```diff
@@ -16,6 +16,9 @@ def forward_to_SOAP_service(json_data, image):
...
+    if plate and type(plate) != str:
+        plate = json_data["data"]["results"][0]["props"]["plate"][0]["value"]
+        score = json_data["data"]["results"][0]["props"]["plate"][0]["score"]
```
Comment (ttayson) [inline webhooks/middleware/protocols/soap.py:21]:
```suggestion
        plate = json_data["data"]["results"][0]["plate"]["props"]["plate"][0]["value"]
        score = json_data["data"]["results"][0]["plate"]["props"]["plate"][0]["score"]
```

File: webhooks/middleware/protocols/synology_rest.py [hunk 3]
```diff
@@ -13,6 +13,8 @@ def process_request(
 
     timestamp = json_data["data"]["timestamp_local"]
     plate = json_data["data"]["results"][0]["plate"]
+    if plate and type(plate) != str:
+        plate = json_data["data"]["results"][0]["props"]["plate"][0]["value"]
```
Comment (ttayson) [inline webhooks/middleware/protocols/synology_rest.py:17]:
```suggestion
        plate = json_data["data"]["results"][0]["plate"]["props"]["plate"][0]["value"]
```

File: webhooks/middleware/protocols/openeye.py [hunk 4]
```diff
@@ -16,6 +16,9 @@ def process_request(
     json_data: dict[str, Any], upload_file: bytes | None = None
 ) -> tuple[str, int]:
     # Prepare the payload for the API request
+    plate = json_data["data"]["results"][0]["plate"]
+    if plate and type(plate) != str:
+        plate = json_data["data"]["results"][0]["props"]["plate"][0]["value"]
```
Comment (ttayson) [inline webhooks/middleware/protocols/openeye.py:21]:
```suggestion
        plate = json_data["data"]["results"][0]["plate"]["props"]["plate"][0]["value"]
```

File: webhooks/middleware/protocols/nx.py [hunk 5]
```diff
@@ -111,6 +111,10 @@ def process_request(
     ssl = os.getenv("SSL", "False").lower() in ("true", "1", "t")
     tag = os.getenv("TAG")
 
+    plate = json_data["data"]["results"][0]["plate"]
+    if plate and type(plate) != str:
+        plate = json_data["data"]["results"][0]["props"]["plate"][0]["value"]
```
Comment (ttayson) [inline webhooks/middleware/protocols/nx.py:116]:
```suggestion
        plate = json_data["data"]["results"][0]["plate"]["props"]["plate"][0]["value"]
```

**Comment by AashishDhakal** [comment]
@ttayson Are you sure about the fields?

I see a different structure here:

Screenshot 2024-10-09 at 08 20 51

**Comment by ttayson** [comment]
@AashishDhakal  Visually it seems correct, but if you add the code to a JSON formatting tool, you can verify it.
We may also need to update the guide with better formatting.

image

**Review by ttayson — APPROVED** [review]
LGTM

**Comment by marcbelmont** [comment]
@dibonjohnseron Please merge if you're also good with this.

**Comment by dibonjohnseron** [comment]
All good to me as well.

**Pull request merged**